CardScanSystemTest/CardScanSystemTestUITests/DenyCameraUITests.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  DenyCameraUITests.swift
+//  CardScanSystemTestUITests
+//
+//  Created by Sam King on 8/29/19.
+//  Copyright © 2019 Sam King. All rights reserved.
+//
+
+import XCTest
+
+class DenyCameraUITests: XCTestCase {
+
+    var app: XCUIApplication?
+    var interruptionToken: NSObjectProtocol?
+    
+    override func setUp() {
+        // this will click both the "deny" option camera permissions and the "ok" button
+        // that explains that the user denied permission
+        self.interruptionToken = addUIInterruptionMonitor(withDescription: "Camera Services") { (alert) -> Bool in
+            alert.buttons.element(boundBy: 0).tap()
+            return true
+        }
+        
+        continueAfterFailure = false
+        self.app = XCUIApplication();
+        self.app?.launch()
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        self.interruptionToken.map { removeUIInterruptionMonitor($0) }
+    }
+
+
+    // The way to test this is to run it on a real device with the CardScanSystemTest
+    // app uninstalled so it can get the camera permission dialog and test it
+    /*
+    func testDeny() {
```

Comment:
And this

Reply:
We should keep this one. It's something that I use periodically to test permissions